src/ExchangeSharp/API/Exchanges/_Base/ExchangeAPI.cs

```diff
@@ -398,75 +400,117 @@ private static IExchangeAPI CreateExchangeAPI(Type? type)
 		/// </summary>
 		/// <typeparam name="T">Type of exchange api to create</typeparam>
 		/// <returns>Created exchange api</returns>
-		public static T CreateExchangeAPI<T>() where T : ExchangeAPI
+		public static async Task<T> CreateExchangeAPIAsync<T>() where T : ExchangeAPI
 		{
-			return (T)CreateExchangeAPI(typeof(T));
+			return (T)await InitializeAPIAsync(typeof(T));
 		}
 
 		/// <summary>
 		/// Get a cached exchange API given an exchange name (see ExchangeName class)
 		/// </summary>
 		/// <param name="exchangeName">Exchange name. Must match the casing of the ExchangeName class name exactly.</param>
 		/// <returns>Exchange API or null if not found</returns>
+		[Obsolete("Use the async version")]
 		public static IExchangeAPI GetExchangeAPI(string exchangeName)
+		{
+			return GetExchangeAPIAsync(exchangeName).Result;
+		}
+
+		/// <summary>
+		/// Get a cached exchange API given an exchange name (see ExchangeName class)
+		/// </summary>
+		/// <param name="exchangeName">Exchange name. Must match the casing of the ExchangeName class name exactly.</param>
+		/// <returns>Exchange API or null if not found</returns>
+		public static Task<IExchangeAPI> GetExchangeAPIAsync(string exchangeName)
 		{
 			Type type = ExchangeName.GetExchangeType(exchangeName);
-			return GetExchangeAPI(type);
+			return GetExchangeAPIAsync(type);
 		}
 
 		/// <summary>
 		/// Get a cached exchange API given a type
 		/// </summary>
 		/// <typeparam name="T">Type of exchange to get</typeparam>
 		/// <returns>Exchange API or null if not found</returns>
+		[Obsolete("Use the async version")]
 		public static IExchangeAPI GetExchangeAPI<T>() where T : ExchangeAPI
+		{
+			return GetExchangeAPIAsync<T>().Result;
+		}
+
+		public static Task<IExchangeAPI> GetExchangeAPIAsync<T>() where T : ExchangeAPI
```

Comment:
Can you update the example in README.md to this new method?

Reply:
Done